src/main/java/org/terasology/scenario/internal/systems/RegionTreeSystem.java

```diff
@@ -210,8 +253,12 @@ public void onRegionResizeEvent(RegionResizeEvent event, EntityRef entity, Scena
 
         scenarioEntity.saveComponent(scenarioEntity.getComponent(ScenarioComponent.class));
         for (EntityRef e : entityManager.getEntitiesWith(ScenarioHubToolUpdateComponent.class)) {
-            e.getComponent(ScenarioHubToolUpdateComponent.class).dirtyRegions = true;
             e.saveComponent(e.getComponent(ScenarioHubToolUpdateComponent.class));
         }
+
+        for (EntityRef e : entityManager.getEntitiesWith(ScenarioRegionVisibilityComponent.class)) {
+            e.getComponent(ScenarioRegionVisibilityComponent.class).dirtyRegionsDraw = true;
+            e.saveComponent(e.getComponent(ScenarioRegionVisibilityComponent.class));
+        }

```

Comment:
this is a repeating block that could be extracted in a method. might not matter anyway when you change the code though.